pkg/machinery/resources/runtime/oom_action.go

```diff
@@ -0,0 +1,72 @@
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+package runtime
+
+import (
+	"time"
+
+	"github.com/cosi-project/runtime/pkg/resource"
+	"github.com/cosi-project/runtime/pkg/resource/meta"
+	"github.com/cosi-project/runtime/pkg/resource/protobuf"
+	"github.com/cosi-project/runtime/pkg/resource/typed"
+
+	"github.com/siderolabs/talos/pkg/machinery/proto"
+)
+
+// OOMActionType is the type of the OOM action record resource.
+const OOMActionType = resource.Type("OOMActions.talos.dev")
+
+// OOMAction is the OOM action record resource.
+type OOMAction = typed.Resource[OOMActionSpec, OOMActionExtension]
+
+// OOMActionSpec describes the OOM action record resource properties.
+//
+//gotagsrewrite:gen
+type OOMActionSpec struct {
+	Time           time.Time      `yaml:"time" protobuf:"1"`
```

Comment:
we could probably skip this, we have `.Metadata().CreatedAt()` which is basically same?